main.go

```diff
@@ -277,7 +281,64 @@ func (p *prometheus) Collect(ch chan<- registry.Metric) {
 	}
 }
 
+func usage() {
+	groups := make(map[string][]*flag.Flag)
+	// Set a default group for ungrouped flags.
+	groups["."] = make([]*flag.Flag, 0)
+
+	// Bucket flags into groups based on the first of their dot-separated levels.
+	flag.VisitAll(func(fl *flag.Flag) {
+		parts := strings.SplitN(fl.Name, ".", 2)
+		if len(parts) == 1 {
+			groups["."] = append(groups["."], fl)
+		} else {
+			name := parts[0]
+			groups[name] = append(groups[name], fl)
+		}
+	})
+
+	groupsOrdered := make(sort.StringSlice, 0, len(groups))
+	for groupName := range groups {
+		groupsOrdered = append(groupsOrdered, groupName)
+	}
+	sort.Sort(groupsOrdered)
+
+	fmt.Fprintf(os.Stderr, "Usage: %s [options ...] -config.file=<config_file>:\n\n", os.Args[0])
+
+	const (
+		maxLineLength = 90
```

Comment:
Ideally it would adapt to the actual terminal size - but calling `stty` introduces more potential for incompatibility.

Also typo on cleanup. Thanks.